src/routes.js

```diff
@@ -157,22 +157,27 @@ module.exports = (
   
   // Hack to check whether or not a wallet exists
   const walletExists = async () => {
-    const availableService = await getAvailableService();
-    if (availableService.service === "lightning") {
-      return true;
-    }
-
-    if (availableService.service === "walletUnlocker") {
-      const randomPassword = Crypto.randomBytes(4).toString('hex');
-      try {
-        await unlockWallet(randomPassword);
+    try {
+      const availableService = await getAvailableService();
+      if (availableService.service === "lightning") {
         return true;
-      } catch (err) {
-        if (err.message.indexOf("invalid passphrase") > -1) {
+      }
+  
+      if (availableService.service === "walletUnlocker") {
+        const randomPassword = Crypto.randomBytes(4).toString('hex');
+        try {
+          await unlockWallet(randomPassword);
           return true;
+        } catch (err) {
+          if (err.message.indexOf("invalid passphrase") > -1) {
+            return true;
+          }
+          return false;
```

Comment:
Could this be a false negative?

Reply:
line 175

Reply:
No, if an error occurred while trying to unlock the wallet and the password isn't wrong, then that must be caused by the wallet not being initialized yet